concordium-consensus/src/Concordium/GlobalState/Persistent/Cache.hs

```diff
@@ -41,6 +48,12 @@ class Cache c where
 
     putCachedValue :: (MonadCache c m) => Proxy c -> CacheKey c -> CacheValue c -> m (CacheValue c)
     lookupCachedValue :: (MonadCache c m) => Proxy c -> CacheKey c -> m (Maybe (CacheValue c))
+    getCacheSize :: (MonadCache c m) => Proxy c -> m Int
+
+newtype CacheContext c = CacheContext { theCacheContext :: c }
```

Comment:
At some point we need to start documenting our abstractions - but I suppose this can be postponed until the integration task has been completed.

Reply:
done